File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    maxDepthFound: number = 0
    /**
     * @param {TreeNode} root
     * @return {number}
     */
    maxDepth(root: TreeNode | null) {
        if (root === null) return 0
        this.findMaxDepth(root, 1)
        return this.maxDepthFound
    }

    findMaxDepth(node: TreeNode, currDepth: number) {
        this.maxDepthFound = Math.max(this.maxDepthFound, currDepth)
        if (node.left !== null) {
            this.findMaxDepth(node.left, currDepth + 1)
        }
        if (node.right !== null) {
            this.findMaxDepth(node.right, currDepth + 1)
        }
    }
}

File: Data Structures & Algorithms/same-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        if not p and not q:
            return True
            
        is_same_tree = False
        if not p or not q:
            is_same_tree = p == q
        else:
            is_same_tree = p.val == q.val

        if not is_same_tree:
            return False
        
        return self.isSameTree(p.left, q.left) and self.isSameTree(p.right, q.right)
        
       